### PR TITLE
Add Postgresql as a storage option

### PR DESCRIPTION
**Pull request by jframe**
Adds PostgreSQL as a storage option
* Also adds openJPA so that other SQL databases can be more easily added in the future
* DDL script to create the database table

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -44,6 +52,17 @@ jobs:
...
+      - run:
+          name: Waiting for Postgres to be ready
+          command: |
+            for i in `seq 1 10`;
+            do
+              nc -z localhost 5432 && echo Success && exit 0
+              echo -n .
+              sleep 1
+            done
+            echo Failed waiting for Postgres && exit 1
```
Comment (CjHare) [inline .circleci/config.yml]: Error message could have greater clarity.

Id this waiting for Postgres to start? to become responsive?

File: build.gradle [hunk 2]
```diff
@@ -13,6 +13,15 @@
 
 import net.ltgt.gradle.errorprone.CheckSeverity
 
+buildscript {
+  repositories { jcenter() }
+
+  dependencies {
+    classpath 'org.apache.openjpa:openjpa:3.1.0'
```
Comment (CjHare) [inline build.gradle]: Why not have these version in `versions.grade` with the other dependency versions?

File: build.gradle [hunk 3]
```diff
@@ -261,6 +277,10 @@ dependencies {
   // XML
   compile 'javax.xml.bind:jaxb-api:2.3.0'
 
+  // database
+  compile 'org.apache.openjpa:openjpa:3.1.0'
```
Comment (CjHare) [inline build.gradle]: Couldn't all these dependency version be in `versions.gradle`?

File: src/main/java/net/consensys/orion/cmd/Orion.java [hunk 4]
```diff
@@ -530,13 +530,9 @@ private KeyValueStore createStorage(String storage, Path storagePath) {
...
       }
     } else if (storage.toLowerCase().startsWith("sql")) {
-      try {
-        // FIXME: Hack to enable update in SQL.
-        // TODO: Update net.consensys.cava.kv.SQLKeyValueStore to enable updates
-        return new OrionSQLKeyValueStore(db);
-      } catch (IOException | SQLException e) {
-        throw new OrionStartException("Couldn't create SQL-backed store: " + db, e);
-      }
+      // FIXME: Hack to enable update in SQL.
```
Comment (CjHare) [inline src/main/java/net/consensys/orion/cmd/Orion.java]: Loose the FIXME

File: .circleci/config.yml (hunk 1 above)
Comment (jframe) [reply .circleci/config.yml]: Done.
It's waiting for Postgres to start and be ready accepting requests on its listening port.

File: build.gradle (hunk 2 above)
Comment (jframe) [reply build.gradle]: Done.
There wasn't one. But have added one.

File: build.gradle (hunk 3 above)
Comment (jframe) [reply build.gradle]: Done.
Didn't exist, but it does now

File: src/main/java/net/consensys/orion/cmd/Orion.java (hunk 4 above)
Comment (jframe) [reply src/main/java/net/consensys/orion/cmd/Orion.java]: Done

**Pull request merged**